### PR TITLE
Test unity analytics

### PR DESCRIPTION
**Pull request by Eryngii01**
Imported UGS Analytics into the project and created custom events to log into the analytics system, in addition to simple reports on the Analytics dashboard:
- **deployTrap** [parameters: string **trapType**] -> Triggered when the player successfully deploys a trap
- **heroDied** [parameters: int **liveLeft**, float **xPos**, float **yPos**, float **zPos**] -> Triggered when the hero dies from the depletion of health or falling out of bounds
- **heroIsStuck** [parameters: float **xPos**, float **yPos**, float **zPos**] -> Triggered when the hero is suspected to have gotten stuck in part of the level
- **switchTrap** [parameters: bool **canAfford**, string **trapType**] -> Triggered when the player switches between traps; might remove this if we don't need it, since it can get quite expensive with firing off events...
- **touchBoundaryDeath** [parameters: int **coinBalance**, float **xPos**, float **yPos**, float **zPos**] -> Triggered if the player loses the game by coming in contact with the camera boundary
- **touchHeroDeath** [parameters: int **coinBalance**, float **xPos**, float **yPos**, float **zPos**] -> Triggered if the player loses the game by coming in contact with the hero

**Note: Currently, the UGS_Analytics module is activated upon the Start method when the MainMenu scene is loaded; however, it collects data immediately before asking for consent, so this will need to be handled in a future update...

### CODE REVIEW

File: Assets/Code/Scripts/UGSAnalytics/UGS_Analytics.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+       try 
+       {
+            await UnityServices.InitializeAsync();
+       } catch (ConsentCheckException e) {
+            Debug.Log(e.ToString());
+       }
+
+       // We want analytics to persist throughout the lifecycle of the entire game so we can trigger its functions
+       // But this means testing must be done by starting with the MainMenu scene
+       DontDestroyOnLoad(transform.gameObject);
```
Comment (eon704) [inline Assets/Code/Scripts/UGSAnalytics/UGS_Analytics.cs:19]: Perhaps we an make it singleton? 😀
Also, what happens if we navigate back to MainMenu scene later on, does it create a second instane of the `UGS_Analytics`?

**Review by eon704 — COMMENTED** [review]
Really nice feature, thank you for adding it! Looks very interesting and promising!
A small neatpick on the `DontDestroyOnLoad` call, just wanted to make sure we don't create multiple instances of the same object.

Also appreciate the **Stuck** logic on the Hero! We needed that

File: Assets/Code/Scripts/UGSAnalytics/UGS_Analytics.cs (hunk 1 above)
Comment (Eryngii01) [reply Assets/Code/Scripts/UGSAnalytics/UGS_Analytics.cs:19]: I scripted it according to the Unity documentation/tutorial, but I'll try to test out what happens if we destroy the UGS_Analytics object between the scenes and also going back to the MainMenu scene. If it works, I'll change this to a singleton as one of the manager classes!

Comment (Eryngii01) [reply Assets/Code/Scripts/UGSAnalytics/UGS_Analytics.cs:19]: It turns out the UGS Analytics connection still stays open regardless of the destruction of the Analytics script, so a singleton works!

**Review by eon704 — APPROVED** [review]
LGTM!

**Pull request merged**